code/modules/paperwork/clipboard.dm

```diff
@@ -72,16 +72,26 @@
 			return
 		push_paper(W)
 		to_chat(user, SPAN_NOTICE("You clip the [W] onto \the [src]."))
+		if(user.machine)
+			attack_self(user)
```

Comment:
Currently, that's the way most things that don't use nano check if they got someone using them. Its pretty awful, but that's the only way you can do it currently. 
I had a rework for properly handling users that was a lot less gross in the works, but I didn't really get around to finishing it, and this PR is already enormous, soooo.

Reply:
If you need to actually check this, store a weakref to the user in CouldUseTopic or interact, unwrap here, listen for on-close and remove the weakref there. But generally there is no need to check for this, you can format the interaction flow to just not need to know whether the UI is open.

The absolute bare minimum here is to check `user.machine == src`, checking `user.machine` is meaningless. But I'd encourage just reconsidering these checks in general.

Reply:
Yeah, its true not checking if user.machine == src is a mistake here..

But, yeah, about the weakref, that's what I did in the rework for interface users. But, the problem is, if I write a whole new custom system for this specific thing it'll just convolute things even more.

And, you can't format interactions to account for external factors like adding a page, dropping the item, etc.. You absolutely need the system I'm currently using here for that (althought I should really be using UpdateUsrDialog() instead of calling attack_self())